Objects/typeobject.c

```diff
@@ -3119,15 +3403,24 @@ PyType_FromSpec(PyType_Spec *spec)
 void *
 PyType_GetSlot(PyTypeObject *type, int slot)
 {
-    if (!_PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE) || slot < 0) {
+    const PySlot_Offset *cur_offset;
+    void *parent_slot;
+
+    if (slot < 0) {
         PyErr_BadInternalCall();
         return NULL;
     }
-    if ((size_t)slot >= Py_ARRAY_LENGTH(slotoffsets)) {
-        /* Extension module requesting slot from a future version */
-        return NULL;
+
+    for (cur_offset = pyslot_offsets; cur_offset->slot; cur_offset++) {
```

Comment:
It would be better to index the array directly; slot numbers are small integers.